test/letters/containers/AddressSection.unit.spec.jsx

```diff
@@ -52,40 +50,39 @@ window.dataLayer = [];
 describe('<AddressSection>', () => {
   // we expect a render with default props to show the AddressContent component
   it('should display an address if one is provided in props', () => {
-    const tree = SkinDeep.shallowRender(<AddressSection {...defaultProps}/>);
-    const addressContent = tree.dive(['AddressContent']);
-    const contentHeader = addressContent.subTree('p').text();
+    const tree = shallow(<AddressSection {...defaultProps}/>);
+    const addressContent = tree.find('AddressContent');
+    const contentHeader = addressContent.dive().find('p').text();
 
     expect(contentHeader).to.contain('Downloaded documents will list your address as:');
   });
 
   it('should display a loading spinner if address save in progress', () => {
     const newProps = { ...defaultProps, savePending: true };
-    const tree = SkinDeep.shallowRender(<AddressSection {...newProps}/>);
-    const spinner = tree.dive(['AddressContent', 'AddressBlock', 'LoadingIndicator']);
+    const tree = shallow(<AddressSection {...newProps}/>);
+    const spinner = tree.find('AddressContent').dive()
+      .find('AddressBlock').dive()
+      .find('LoadingIndicator')
+      .dive();
```

Comment:
Yeah that's definitely more verbose, eh? I don't see a way around this at the moment. Maybe this is a sign that the component hierarchy could be refactored in a way that makes each level easier to test. 🤷‍♂️ Interesting thought for the future.

Reply:
Actually this is interesting...looks as though you don't have to go level-by-level. This test also works when `spinner` shortened like so:
`const spinner = tree.find('LoadingIndicator').dive();`

Reply:
Ah OK, so I was wondering why this skip-level works here but not elsewhere. I guess it's because `LoadingIndicator` is actually explicitly passed in from `AddressSection` whereas other subcomponents (e.g., `AddressBlock`) are rendered by the child components of `AddressSection`. That is, `AddressSection` knows about `LoadingIndicator` but not about things like `AddressBlock` that are called in `AddressContent`. I feel like there's room to standardize our approach here, by passing the `isPending` (or whatever it's called) prop to `AddressContent`, and let that component handle the logic, as opposed to keeping all that logic in `AddressSection` and passing down components as children. Not something we need to mess with (and especially not in this PR...) but I feel like it's worth calling out for posterity.

Reply:
I feel like if we're having dive down multiple times, the test isn't a good fit for shallow rendering in general.

Reply:
Yeah, that's probably fair. I think we were only doing shallow rendering in the first place to avoid rendering all the _other_ stuff that we weren't drilling down into, but it's really kinda whatever. I'm a fan of just `mount`ing the silly thing if we need to `dive` deep. 👍